docs/guides/moodleapp/development/plugins-upgrade-guide.md

```diff
@@ -0,0 +1,135 @@
+- The function `syncOnSites` from `CoreSyncBaseProvider` now expects to receive a function with the parameters already bound:
+
+```javascript title="Ionic 3"
+    syncOnSites('events', this.syncAllEventsFunc.bind(this), [siteId);
+```
+
+```javascript title="Ionic 5"
+    syncOnSites('events', this.syncAllEventsFunc.bind(this, force), siteId);
+```
+
+- All the delegates that previously supplied an injector parameter to its handlers no longer do that. For example, the function `getComponent()` in `CoreUserProfileFieldDelegate` used to receive an injector as a parameter, but now it won’t receive any parameter.
+- All the delegates that previously supplied a `NavController` parameter to its handlers no longer do that. For example, the function `openCourse()` in `CoreCourseFormatDelegate` no longer receive the `NavController` parameter.
+- The handlers registered in `CoreCourseOptionsDelegate` now need to return the properties `page` and `pageParams` instead of `component` and `componentData`. Please notice this only affects your plugin if you’re creating the handler yourself using JavaScript code.
+- The handlers registered in `CoreUserDelegate` have changed a bit. Please notice this only affects your plugin if you’re creating the handler yourself using JavaScript code.
+  - Handlers can now define a `cacheEnabled` property (`false` by default) to cache `isEnabledForUser` calls.
+  - In the function `isEnabledForUser`, the `navOptions` and `admOptions` parameters have been removed.
+  - `isEnabledForUser` is now optional and defaults to `true`.
+  - They can implement a new function called `isEnabledForCourse`; this function will receive the `navOptions` and `admOptions` parameters. If it's not defined, it'll default to `true`.
+- The function `prefetchPackage` in the `CoreCourseActivityPrefetchHandlerBase` has changed. If you were using this class to implement your own prefetch handler you might need to update its code.
+- `CoreInitDelegate` has been deleted. Now the initialisation of the app is done via Angular’s `APP_INITIALIZER`. Please notice that `APP_INITIALIZER` cannot and shouldn’t be used by plugins.
+- The function `getAdditionalDownloadableFiles` in question types now needs to return a list of `CoreWSExternalFile`, it no longer accepts a list of strings.
+- Files stored to be uploaded later using `CoreFileUploaderProvider` no longer have an `offline` property, now they’re just instances of `FileEntry`.
+- `ionViewCanLeave` function has been renamed to `canLeave`.
+- The `onchange` method of the `Network` service is now called `onChange`.
+
+## Supporting both Ionic 3 and Ionic 5
+
+Your plugin should still support Ionic 3 so it works in devices that haven’t updated the app yet. This can be done by checking the value of `appversioncode` sent by the app.Here you can find an example applied to the `choicegroup` plugin: [Choicegroup plugin](https://github.com/dpalou/moodle-mod_choicegroup/blob/ionic5/classes/output/mobile.php#L52).
```

Comment:
```suggestion
Your plugin should still support Ionic 3 so it works on devices that haven't updated the app yet. This can be done by checking the value of `appversioncode` sent by the app. Here you can find an example applied to the `choicegroup` plugin: [Choicegroup plugin](https://github.com/dpalou/moodle-mod_choicegroup/blob/ionic5/classes/output/mobile.php#L52).
```